doc/Development_Documentation/10_E-Commerce_Framework/07_Filter_Service/03_Elastic_Search.md

```diff
@@ -1,52 +1,180 @@
+                        template: ':Shop/filters:select.html.php'
+
+                    FilterMultiSelect:
+                        filter_type_id: Pimcore\Bundle\EcommerceFrameworkBundle\FilterService\FilterType\ElasticSearch\MultiSelect
+                        template: ':Shop/filters:multiselect.html.php'
+
+                    FilterMultiSelectFromMultiSelect:
+                        filter_type_id: Pimcore\Bundle\EcommerceFrameworkBundle\FilterService\FilterType\ElasticSearch\MultiSelectFromMultiSelect
+                        template: ':Shop/filters:multiselect.html.php'
+
+                    FilterMultiRelation:
+                        filter_type_id: Pimcore\Bundle\EcommerceFrameworkBundle\FilterService\FilterType\ElasticSearch\MultiSelectRelation
+                        template: ':Shop/filters:multiselect-relation.html.php'
+
+                    FilterCategory:
+                        filter_type_id: Pimcore\Bundle\EcommerceFrameworkBundle\FilterService\FilterType\ElasticSearch\SelectCategory
+                        template: ':Shop/filters:select_category.html.php'
+
+                    FilterRelation:
+                        filter_type_id: Pimcore\Bundle\EcommerceFrameworkBundle\FilterService\FilterType\ElasticSearch\SelectRelation
+                        template: ':Shop/filters:object_relation.html.php'
+```
+
+## Filter for nested documents
+
+In some cases it is necessary to store an array of objects, but in a way so that they can be queried independently of each other, i.e. if you want 
+to store the keys of a classification store dataobject field. The data in your index may look as follows:
+
+```json
+{
+   "_source": {
+       "attributes": {  
+         "my_attributes": [  
+            {  
+               "id": "12345",
+               "name": "Höhe",
+               "value": "15 mm"
+            },
+            {  
+               "id": "12346",
+               "name": "Länge",
+               "value": "7 mm"
+            },
+            {  
+               "id": "12347",
+               "name": "Breite",
+               "value": "30 mm"
+            }
+         ]
+      }
+   }
+}
+```
+
+To utilize the `nested` document functionality the mapping type of the field `my_attributes` must be defined as `nested`, to let elastic search know about the sub-documents:
+
+```yaml
+ attributes:
+    filtergroup: string
+    type: 'nested'
+    options:
+        mapping:
+            type: 'nested'
+    interpreter_id: AppBundle\Ecommerce\IndexService\Interpreter\MyAttributes
+    interpreter_options:
+        locale: '%%locale%%'
+```
+
+Now you can create a filter for the nested document field, which has to be defined in a nested manner as well:
+
+```php
+class SelectMyAttribute extends \Pimcore\Bundle\EcommerceFrameworkBundle\FilterService\FilterType\ElasticSearch\AbstractFilterType
```

Comment:
there is no such thing as \Pimcore\Bundle\EcommerceFrameworkBundle\FilterService\FilterType\ElasticSearch\AbstractFilterType in the framework. 
Would that be needed? Maybe missing in the PR?

Reply:
how would the `getFilterFrontend` method look like?

Reply:
Thx, fixed the namespace. I also added an example implementation for the getFilterFrontend().